cogs/audio.py

```diff
@@ -1491,26 +1491,26 @@ def _clean_url(self, url):
 
         # Checking already connected, will join if not
 
-        try:
-            self.has_connect_perm(author, server)
-        except AuthorNotConnected:
-            await self.bot.say("You must join a voice channel before I can"
-                               " play anything.")
-            return
-        except UnauthorizedConnect:
-            await self.bot.say("I don't have permissions to join your"
-                               " voice channel.")
-            return
-        except UnauthorizedSpeak:
-            await self.bot.say("I don't have permissions to speak in your"
-                               " voice channel.")
-            return
-        except ChannelUserLimit:
-            await self.bot.say("Your voice channel is full.")
-            return
-
         if not self.voice_connected(server):
-            await self._join_voice_channel(voice_channel)
+            try:
+                self.has_connect_perm(author, server)
```

Comment:
This line right here (well, specifically the full channel comparison in the function call (https://github.com/Cog-Creators/Red-DiscordBot/blob/97b70c35ee540710d7172e4fa7f6ac3663a979ea/cogs/audio.py#L2246) is the root cause of the issue, as there is no offset if the bot is in the channel, nor an additional pass condition of the bot is in the channel already with speak perms.